docs/openapi-docs/src/main/scala/sttp/tapir/docs/openapi/EndpointToOpenAPIPaths.scala

```diff
@@ -44,6 +45,15 @@ private[openapi] class EndpointToOpenAPIPaths(
     (e.showPathTemplate(showQueryParam = None, includeAuth = false, showNoPathAs = "/", showPathsAs = None), pathItem)
   }
 
+  private def filterOutHiddenInputs(inputs: Vector[EndpointInput.Basic[_]]) = inputs.collect {
+    case OneOfBody(variants, mapping) =>
+      OneOfBody(
+        variants.filterNot(_.codec.schema.hidden),
+        mapping
+      )
+    case a: EndpointInput.Atom[_] if !a.codec.schema.hidden => a
```

Comment:
Collect uses PartialFunction, which is prone to overlooking cases. Maybe filter with a total function is safer so that we explicitly see which cases are rejected, guarded by the compiler?